packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -44,15 +50,65 @@ class AndroidPlugin extends PluginPlatform {
   final String name;
   final String package;
   final String pluginClass;
+  final String pluginPath;
 
   @override
   Map<String, dynamic> toMap() {
     return <String, dynamic>{
       'name': name,
       'package': package,
       'class': pluginClass,
+      'usesNewEmbedding': _usesNewEmbedding,
     };
   }
+
+  /// Returns [true] if the plugin is using the new Android embedding.
+  bool get _usesNewEmbedding {
```

Comment:
This is a bit heavyweight for a getter. Also, if the result isn't expected to change, then we can cache the result instead of hitting the filesystem every time. Maybe:
```dart
bool _hasNewEmbeddingFile;
bool get _usesNewEmbedding => _hasNewEmbeddingFile ??= _findNewEmbeddingFile();
bool _findNewEmbeddingFile() {
  ...
}
```
Or with different names than these that make more sense.

Reply:
Good call. Thanks